internal/pipe/changelog/changelog.go

```diff
@@ -60,55 +72,11 @@ func (Pipe) Run(ctx *context.Context) error {
 		return err
 	}
 
-	changelogStringJoiner := "\n"
-	if ctx.TokenType == context.TokenTypeGitLab || ctx.TokenType == context.TokenTypeGitea {
-		// We need two or more whitespace to let markdown interpret
-		// it as newline. See https://docs.gitlab.com/ee/user/markdown.html#newlines for details
-		log.Debug("is gitlab or gitea changelog")
-		changelogStringJoiner = "   \n"
-	}
-
-	changelogElements := []string{
-		"## Changelog",
-	}
-
-	if ctx.Config.Changelog.Use == "github-native" {
-		changelogElements = []string{strings.Join(entries, changelogStringJoiner)}
-	} else if shouldGroup(ctx.Config.Changelog) {
-		log.Debug("grouping entries")
-		groups := ctx.Config.Changelog.Groups
-
-		sort.Slice(groups, func(i, j int) bool { return groups[i].Order < groups[j].Order })
-		for _, group := range groups {
-			items := make([]string, 0)
-			if group.Regexp == "" {
-				// If no regexp is provided, we purge all strikethrough entries and add remaining entries to the list
-				items = getAllNonEmpty(entries)
-				// clear array
-				entries = nil
-			} else {
-				regex, err := regexp.Compile(group.Regexp)
-				if err != nil {
-					return fmt.Errorf("failed to group into %q: %w", group.Title, err)
-				}
-				for i, entry := range entries {
-					match := regex.MatchString(entry)
-					if match {
-						items = append(items, li+entry)
-						// Striking out the matched entry
-						entries[i] = ""
-					}
-				}
-			}
-			if len(items) > 0 {
-				changelogElements = append(changelogElements, fmt.Sprintf("### %s", group.Title))
-				changelogElements = append(changelogElements, strings.Join(items, changelogStringJoiner))
-			}
-		}
-	} else {
-		log.Debug("not grouping entries")
-		changelogElements = append(changelogElements, strings.Join(getAllNonEmpty(entries), changelogStringJoiner))
+	changes, err := formatChangelog(ctx, entries)
+	if err != nil {
+		return err
 	}
+	changelogElements := append([]string{"## Changelog"}, changes)
```

Comment:
@caarlos0 FYI I purposefully removed the `## Changelog` element when using `github-native`, as this integration already provide second-level headers: https://github.com/goreleaser/goreleaser/blob/ce3e3ed14e67ddee86fbe76ab3e92c68ec4acc00/internal/pipe/changelog/changelog.go#L76

Keeping `## Changelog` for `github-native` would mean the generated log would look like that:

```markdown
## Changelog

<!-- Release notes generated using configuration in .github/release.yml at v2.0.0-beta.6 -->
## What's Changed
### Other changes
* tomljson: port to v2 by @pelletier in https://github.com/pelletier/go-toml/pull/725
* jsontoml: port to v2 by @pelletier in https://github.com/pelletier/go-toml/pull/726
* tomll: port to v2 by @pelletier in https://github.com/pelletier/go-toml/pull/727
**Full Changelog**: https://github.com/pelletier/go-toml/compare/v2.0.0-beta.5...v2.0.0-beta.6
```

Personally I'd need to remove that line for every release. Totally understand it may be a case of personal preference. Just flagging it to you, as I'd expect goreleaser to use the output of the GitHub API as-is.

Reply:
good catch! fixed